packages/toc/src/toc_item.tsx

```diff
@@ -58,8 +180,23 @@ class TOCItem extends React.Component<IProperties, IState> {
       heading.onClick();
     };
 
+    addCommands(commands, toc, heading);
```

Comment:
```suggestion
    addCommands(commands);
```

and move it to constructor so it is only called once. Or better to the `activate()` function of the toc-extension package.